--- a/arch/arm/dts/rk3399.dtsi
+++ b/arch/arm/dts/rk3399.dtsi
@@ -1517,6 +1517,7 @@
 		reg = <0x0 0xff848000 0x0 0x100>;
 		clocks = <&cru PCLK_WDT>;
 		interrupts = <GIC_SPI 120 IRQ_TYPE_LEVEL_HIGH 0>;
+		status = "okay";
 	};
 
 	rktimer: rktimer@ff850000 {
--- a/drivers/watchdog/designware_wdt.c
+++ b/drivers/watchdog/designware_wdt.c
@@ -20,6 +20,11 @@
 #define DW_WDT_CR_RMOD_OFFSET	0x01
 #define DW_WDT_CRR_RESTART_VAL	0x76
 
+#if defined(CONFIG_ROCKCHIP_RK3399)
+#define CONFIG_DW_WDT_BASE	0xff848000
+#define CONFIG_DW_WDT_CLOCK_KHZ	100000
+#endif
+
 struct designware_wdt_priv {
 	void __iomem	*base;
 	unsigned int	clk_khz;
--- a/scripts/Makefile.spl
+++ b/scripts/Makefile.spl
@@ -95,6 +95,7 @@
 libs-y += drivers/
 libs-$(CONFIG_SPL_USB_GADGET) += drivers/usb/dwc3/
 libs-$(CONFIG_SPL_USB_GADGET) += drivers/usb/cdns3/
+libs-$(CONFIG_DESIGNWARE_WATCHDOG) += drivers/watchdog/
 libs-y += dts/
 libs-y += fs/
 libs-$(CONFIG_SPL_POST_MEM_SUPPORT) += post/drivers/
